### DISCUSSION TITLE
@Deprecated(since = "4.1.0") org.eclipse.rdf4j.common.iteration.Iterations

### DISCUSSION THREAD

**Discussion by naturzukunft** [eclipse-rdf4j/rdf4j#4229] [General]
Hi, is there a migration guide?
Thanks Fredy

**Comment by ate47** [comment]
From [Iteration.java](https://github.com/eclipse/rdf4j/blob/main/core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/Iteration.java):

> For performance and simplification the Iteration interface is deprecated and will be removed in 5.0.0. Use CloseableIteration instead, even if your iteration doesn't require AutoCloseable.

From [CloseableIteration.java](https://github.com/eclipse/rdf4j/blob/main/core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/CloseableIteration.java):

> In the future this interface will stop extending {@link Iteration} and instead declare the same interface methods directly. The interface will also stop requiring implementations to automatically close when exhausted, instead making this an optional feature and requiring the user to always call close. This interface may also be removed.

From what I'm understanding, the best you can do is continue using Iterations, replacing all you Iteration to CloseableIteration and when the 5.0.0 will arrive, you will need to replace Iterations by Collections
